exporter/sumologicexporter/filtering.go

```diff
@@ -0,0 +1,101 @@
+package sumologicexporter
+
+import (
+	"fmt"
+	"regexp"
+	"sort"
+	"strconv"
+	"strings"
+
+	"go.opentelemetry.io/collector/consumer/pdata"
+)
+
+type filtering struct {
+	regexes []*regexp.Regexp
+}
+
+// FieldsType represents concatenated metadata
+type FieldsType string
```

Comment:
To be more idiomatic I'd go for

```suggestion
// Fields represents concatenated metadata
type Fields string
```